### DISCUSSION TITLE
What is the decision-making process for choosing between the remote module and the host module when module is shared?

### DISCUSSION THREAD

**Discussion by tachish** [webpack/webpack#17909] [Q&A]
Here is the reproducible repo (`node version >= 16`) 👉 https://github.com/tachish/ModuleFederationSharedDiscussion

**Boost Remote Project:**
```
cd ./host
npm i
npm run start
```
**Boost Host Project:**
```
cd ./remote
npm i
npm run start
```

Host Project: `http://localhost:3000/`
Remote Project: `http://localhost:3001/`

### Background:
Host project shares `vue` and `lodash` with remote project, only using configuration `singleton: true`. Versions of `vue` and `lodash` are literally same in host and remote.

I notice that host project loads remote project's `vue` instead of loading itself's, as following:
![image](https://github.com/webpack/webpack/assets/37827090/ac57cdba-fb00-45bd-bf05-765333933754)
using `vue` from remote project (host: `http://localhost:3001`)

### Question:
I'm curious about the decision-making process between using the host module or the remote module. I'm unsure if this is determined by semver, or if the module loading sequence is initiated by the host project or the remote project.

**Comment by alexander-akait** [comment]
Why you use https://github.com/tachish/ModuleFederationSharedDiscussion/blob/main/host/webpack.config.js#L36?
